Documentation/contributing/development/datapath_config.rst

```diff
@@ -0,0 +1,302 @@
+    ...
+    // The endpoint's security ID.
+    EndpointID uint16 `config:"endpoint_id"`
+    ...
+  }
+
+As shown in the preceding snippet, the new struct field carries our helpful
+comment we provided in the C code and refers to the ``endpoint_id`` variable we
+declared.
+
+.. note::
+
+  At the time of writing, populating Go configuration scaffolding still mostly
+  happens in ``pkg/datapath/loader`` and is scattered between a few places. The
+  goal is to create StateDB tables for each configuration object. These can be
+  managed from Hive Cells and automatically trigger a reload of the necessary
+  BPF programs when any of the values change. This document will be updated
+  along with these changes.
+
+Now, we need to wire up the field with an actual value. Depending on which
+object you're adding configuration to and depending on whether it's 'node
+configuration' (more below) or object-specific, you may need to look in different
+places. For example, adding a value to ``bpf_lxc.c`` like in this example, the
+value is typically set in ``endpointRewrites()``:
+
+.. code-block:: go
+
+  func endpointRewrites(...) ... {
+    ...
+    cfg.InterfaceIfindex = uint32(ep.GetIfIndex())
+    ...
+  }
+
+.. warning::
+
+  This plumbing needs to be done for every object that needs access to the
+  variable! For example, if you declare a variable in a header common to both
+  ``bpf_lxc.c`` and ``bpf_host.c``, you'll need to make sure the agent supplies
+  the value to both structs.
+
+If, for some reason, this document no longer matches the codebase, grep around
+for uses of the various structs and their fields, and extend the existing code.
+Over time, Hive Cells will be able to write to these structs using StateDB
+tables.
+
+Reading the Variable in C
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We've declared our global config variable. We've generated Go code and wired up
+a value from the agent. Now, we need to put the variable to use!
+
+In datapath bpf code, we can refer to it using the ``CONFIG()`` macro. This
+macro resolves to a special variable name representing our configuration value,
+which could change in the future. The macro is there to avoid cross-cutting code
+changes if we ever need to make changes here.
+
+.. note::
+
+  The variable is not a compile-time constant, so it cannot be used to control
+  things like BPF map sizes or to initialize other global ``const`` variables at
+  compile time.
+
+.. code-block:: C
+
+  CONFIG(endpoint_id)
+
+Use the macro like you would typically use a variable:
+
+.. code-block:: c
+
+  __u16 endpoint_id = CONFIG(endpoint_id);
+
+or in a branch:
+
+.. code-block:: c
+
+  if (CONFIG(endpoint_id) != 0) {
+    ...
+  }
+
+Node Configuration
+~~~~~~~~~~~~~~~~~~
+
+.. warning::
+
+  Historically, most of the agent's configuration was presented to the datapath
+  as 'node configuration' (in ``node_config.h``), but this pattern is
+  discouraged going forward and may go away at some point in the future. More on
```

Comment:
Are there any CI checks in place to catch new occurrences of “node configuration” changes?

Reply:
For now, we have all relevant files assigned to `@cilium/loader`, so a small group of in-the-loop folks need to sign off on any changes. There are warning comments littered throughout to alert contributors and direct them to the docs. It's all relatively chaotic code, so I can't think of an effective linter that's easy to implement. For now, we'll direct contributors to the docs if anything shows up in a PR, and we can always lend a hand or make suggestions if needed.

Reply:
OK, I understand. Thanks!